runtime/v2/runc/task/service.go

```diff
@@ -201,12 +220,29 @@ func (s *service) Start(ctx context.Context, r *taskAPI.StartRequest) (*taskAPI.
 			Pid:         uint32(p.Pid()),
 		})
 	}
-	s.eventSendMu.Unlock()
+	s.handleStarted(container, p)
 	return &taskAPI.StartResponse{
 		Pid: uint32(p.Pid()),
 	}, nil
 }
 
+func (s *service) handleStarted(container *runc.Container, p process.Process) {
+	pid := p.Pid()
+	s.lifecycleMu.Lock()
+	defer s.lifecycleMu.Unlock()
+	if ees, ok := s.earlyExits[pid]; ok {
```

Comment:
Trying to understand the change: Assuming that the `p` is container-type and we failed to start it. `p.Pid()` is non-zero. If the `processExits()` receives event and locks the `s.lifecycleMu` first, the `handleStarted` will handle the `handleProcessExit`. Is it correct? If so, as far as I know, the worse case is that when the `runc kill -all` takes long time, the `earlyExitsGC` is trying to lock the `earlyExitsGCMu`, which will block any incoming `Start` request even if the lock hasn't been write-locked yet.

Reply:
will pid reuse?